integration/test/upgradetiller/upgrade_tiller_test.go

```diff
@@ -18,19 +19,55 @@ func TestUpgradeTiller(t *testing.T) {
 
 	var err error
 
-	// Install tiller with the latest image.
-	err = config.HelmClient.EnsureTillerInstalled(ctx)
+	labelSelector := "app=helm,name=tiller"
+	outdatedTillerImage := "gcr.io/kubernetes-helm:v2.7.2"
+	tillerNamespace := "giantswarm"
+
+	var helmClient *helmclient.Client
+	{
+		c := helmclient.Config{
+			K8sClient: config.K8sClient,
+			Logger:    config.Logger,
+
+			RestConfig:      config.RestConfig,
+			TillerImage:     outdatedTillerImage,
+			TillerNamespace: tillerNamespace,
+		}
+
+		helmClient, err = helmclient.New(c)
+		if err != nil {
+			t.Fatalf("could not create tiller client %#v", err)
+		}
+	}
+
+	// Install tiller using helm client with outdated image.
+	err = helmClient.EnsureTillerInstalled(ctx)
 	if err != nil {
 		t.Fatalf("could not install tiller %#v", err)
 	}
 
-	namespace := "giantswarm"
-	labelSelector := "app=helm,name=tiller"
+	tillerImage, err := getTillerImage(ctx, tillerNamespace, labelSelector)
+	if err != nil {
+		t.Fatalf("could not get tiller image %#v", err)
+	}
+	if tillerImage != outdatedTillerImage {
+		t.Fatalf("tiller has not been downgraded got image %#q expected %#q", tillerImage, outdatedTillerImage)
+	}
 
-	_, err = getTillerImage(ctx, namespace, labelSelector)
+	// Upgrade tiller to the latest image using default helm client.
```

Comment:
I think `config.HelmClient.tillerImage = hlemclient.TillerImageSpec` line is missing here.